test/e2e/e2e_test.go

```diff
@@ -279,7 +279,8 @@ func TestListRecords(t *testing.T) {
 			t.Fatal(err)
 		}
 
-		if diff := cmp.Diff(want.Records, got.Records, protocmp.Transform()); diff != "" {
+		// Compare only record names. Comparing records data is susceptable to race conditions.
+		if diff := cmp.Diff(recordNames(t, want.Records), recordNames(t, got.Records), protocmp.Transform()); diff != "" {
```

Comment:
@alan-ghelardi want to check with you if this is acceptable. I believe this is the intent of your test?

Reply:
I'm wondering whether this race condition is exactly what this [pull request](https://github.com/tektoncd/results/pull/298) is fixing. Anyhow, comparing only the record names looks good to me. After merging both prs, I'd like to write a test case to cover that bug.
/lgtm

Reply:
IMO the race condition I'm observing is not related. Basically the e2e tests in CI take much longer (6+ mins compared to 2+ mins locally), so when the TaskRun and PipelineRun are called at the beginning of the e2e tests, then doing 2 ListRecords calls later was getting small differences in the data (mainly container state and timestamps) as things are still running. I was able to reproduce this locally by introducing 1-2 seconds delay between the 2 ListRecords calls and I was also able to eliminate the issue with introducing long enough delay after the TaskRun and PipelineRun calls and before the ListRecords calls.

Reply:
Makes sense.